aws/resource_aws_transfer_server.go

```diff
@@ -197,6 +231,8 @@ func resourceAwsTransferServerRead(d *schema.ResourceData, meta interface{}) err
 	endpoint := meta.(*AWSClient).RegionalHostname(fmt.Sprintf("%s.server.transfer", d.Id()))
 
 	d.Set("arn", resp.Server.Arn)
+	d.Set("certificate", resp.Server.Certificate)
+	d.Set("protocols", flattenStringList(resp.Server.Protocols))
```

Comment:
Can use `flattenStringSet` here if `protocols` is changed to `schema.TypeSet`.
Also, please check the error returned by `d.Set()` for non-primitive types.